pkg/reconciler/pipeline/controller.go

```diff
@@ -161,6 +164,9 @@ func (r *pipelineReconciler) reconcile(ctx context.Context, pl *dfv1.Pipeline) (
 	if err != nil {
 		r.recorder.Eventf(pl, corev1.EventTypeWarning, "ReconcilePipelineFailed", "Failed to reconcile pipeline: %v", err.Error())
 	}
+
+	pl.Status.SetObservedGeneration(pl.Generation)
+
```

Comment:
Sorry for the back and forth... thinking if would be better to add it to the end of `reconcileNonLifecycleChanges()`?

Reply:
done